module/Motion/CorrectingRotation.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file   CorrectingRotation.h
+ * @brief  角度補正回頭
+ * @author aridome222
+ */
+
+#include "CorrectingRotation.h"
+
+using namespace std;
+
+CorrectingRotation::CorrectingRotation(int _targetAngle, double _targetSpeed)
+  : targetAngle(_targetAngle), targetSpeed(_targetSpeed){};
+
+void CorrectingRotation::run()
+{
+  // 事前条件を判定する
+  if(!isMetPrecondition()) {
+    return;
+  }
+
+  /*
+   * rear_camera.shより直線の角度を取得
+   * シミュレータ環境ではpopenが使えないため，
+   * ファイルに出力して読み込む
+   */
+  FILE* fp = fopen("result.txt", "r");
+  char output[8];  // rear_camera.shの出力結果を保持する領域
+  char cmd[1024];
+  snprintf(cmd, 1024, "bash ./etrobocon2023/scripts/rear_camera.sh %d > result.txt",
+           ANGLE_SERVER_PORT);
+  system(cmd);
```

Comment:
bashファイルの名前を変更してほしいです。
別ファイルや通信を経由しているので、出来るだけわかりやすくしたいです。
```suggestion
  snprintf(cmd, 1024, "bash ./etrobocon2023/scripts/rear_camera_request.sh %d > response.txt",
           ANGLE_SERVER_PORT);
```


それと、以下の流れだと認識しているのであってるか教えてください。


(あらかじめサーバを起動)
↓
`rear_camera.sh`を呼び出し、以下の処理でサーバにangleってコマンドを送信する
`bash -c "echo angle | nc 127.0.0.1 ${port}"`
↓
リクエストを受信したサーバが、`rear_camera/__main__.py`で設定された以下の処理で`line_angle_calculator.calc_yaw_angle()`を呼び出す
```
    angle_server = AngleServer(
        server_ip=args.server_ip,
        server_port=args.server_port,
        listen_num=127,
        buf_size=1024,
        debug=args.debug
    )

    angle_server.add_command(
        'angle',
        lambda _: str(line_angle_calculator.calc_yaw_angle())
    )
```
↓
`line_angle_calculator.calc_yaw_angle()`が返した値を`result.txt`にリダイレクトで書き込む
↓
`result.txt`内の値を読み込む

Reply:
その認識で間違いないです。